### PR TITLE
[Refactoring] Remove Type::tboolean and Type::tindex

### PR DESCRIPTION
**Pull request by 9rnsr**

### CODE REVIEW

File: src/expression.c [hunk 1]
```diff
@@ -3696,7 +3696,7 @@ Expression *StringExp::semantic(Scope *sc)
             string = buffer.extractData();
             len = newlen;
             sz = 4;
-            //type = new TypeSArray(Type::tdchar, new IntegerExp(loc, len, Type::tindex));
+            //type = new TypeSArray(Type::tdchar, new IntegerExp(loc, len, Type::tsize_t));
```
Comment (yebblies) [inline src/expression.c]: You should probably just delete these lines, they look like D1 string stuff to me.

Comment (9rnsr) [inline src/expression.c]: OK, I removed these lines.

**Comment by yebblies** [comment]
Auto-merge toggled on

**Comment by 9rnsr** [comment]
Thanks!

**Comment by yebblies** [comment]
No worries.  Do you have any thoughts on #4303?

**Pull request merged**